src/backend/joanie/core/models/products.py

```diff
@@ -1209,6 +1210,49 @@ def init_flow(self, billing_address=None):
 
         self.flow.update()
 
+    def get_date_next_installment_to_pay(self):
+        """Get the next due date of installment to pay in the payment schedule."""
+        return next(
+            (
+                installment["due_date"]
+                for installment in self.payment_schedule
+                if installment["state"] == enums.PAYMENT_STATE_PENDING
+            ),
+            None,
+        )
+
+    def get_count_installments_paid(self):
+        """Get the number of installments paid already in the payment schedule."""
+        return sum(
+            1
+            for installment in self.payment_schedule
+            if installment["state"] == enums.PAYMENT_STATE_PAID
+        )
+
+    def get_remaining_balance_to_pay(self):
+        """Get the amount of installments remaining to pay in the payment schedule."""
+        amounts = (
+            Money(installment["amount"])
+            for installment in self.payment_schedule
+            if installment["state"] == enums.PAYMENT_STATE_PENDING
+        )
+        return Money.sum(amounts)
+
+    def get_position_last_paid_installment(self):
+        """Get the position of the last installment paid in the payment schedule."""
+        last_paid_index = 0
+        for index, entry in enumerate(self.payment_schedule, start=1):
+            if entry["state"] == enums.PAYMENT_STATE_PAID:
+                last_paid_index = index
+        return last_paid_index
```

Comment:
IMO, this method should return `None` if no installment is paid. Then the first index should start at `0`